### ISSUE TITLE
Object keys as default order?

### ISSUE THREAD

**Issue by freehuntx** [arloliu/buffer-plus#6]
I stumbled uppon this library and it looks promising 👍 
What do you think about the order feld having a default value of:

```
  get order() {
    if (!this._order) this._order = Object.keys(this.properties)
    return this._order
  }
```

**Comment by arloliu** [comment]
It's really a good question. Even I tested most of javascript engines likes v8, SpiderMonkey...etc.

It seems the order of object properties are guaranteed, but in older version of EMCAscript spec. does not guarantee the property order.

The related discussion in Stackoverflow is [here](https://stackoverflow.com/questions/30076219/does-es6-introduce-a-well-defined-order-of-enumeration-for-object-properties) and [here](https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order)

This package are developed several years ago, I know it's a bit annoying to add order property, I don't like it neither, but I added it because this encoding format might be exchanged between different language, and some language might not guarantee property order.

For example, I have implemented a golang version of this encoding format(will push to Github next week), and in golang, the order of key in hashmap is not guaranteed.

BTW, I think it might be the time that sets this property from required to optional in this package.

I will modify code to support default ordering ASAP.
